lib/compat.php

```diff
@@ -102,6 +102,37 @@ function gutenberg_shim_fix_api_request_plain_permalinks( $scripts ) {
 }
 add_action( 'wp_default_scripts', 'gutenberg_shim_fix_api_request_plain_permalinks' );
 
+/**
+ * Shims support for emulating HTTP/1.0 requests in wp.apiRequest
+ *
+ * @see https://core.trac.wordpress.org/ticket/43605
+ *
+ * @param WP_Scripts $scripts WP_Scripts instance (passed by reference).
+ */
+function gutenberg_shim_api_request_emulate_http( $scripts ) {
+	$api_request_fix = <<<JS
+( function( wp ) {
+	var oldApiRequest = wp.apiRequest;
+	wp.apiRequest = function ( options ) {
+		if ( options.method ) {
+			if ( [ 'PATCH', 'PUT', 'DELETE' ].indexOf( options.method.toUpperCase() ) >= 0 ) {
```

Comment:
Alternatively, we could inverse the check to better reflect that we support _only_ `GET` and `POST`:

```js
if ( [ 'GET', 'POST' ].indexOf( options.method.toUpperCase() ) === -1 ) {
```

_Aside: Curiosity got me wondering: https://jsperf.com/indexof-array-vs-regexp_ (tl;dr: This is good)

Reply:
`OPTIONS` shouldn't be converted, either. (Neither should `HEAD`, `CONNECT`, or `TRACE`, even though we don't specifically use them anywhere.) I'm inclined to go with the same conversion as Backbone, as that's the one that is most likely to be supported.

Reply:
Hmm. I'm not clear what exactly we're doing in this pull request then. Are we targeting HTTP/1.0 ? There are only 3 defined verbs (previously overlooked `HEAD`):

https://www.w3.org/Protocols/HTTP/1.0/spec.html#Methods

It makes me a bit nervous that we target Backbone, as it's not obvious _why_ we target Backbone (aside from anecdotal evidence that it appears to be well-supported).